src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -76,6 +81,9 @@ const Embedding = (props) => {
   const [selectedIds, setSelectedIds] = useState(new Set());
   const [cellColors, setCellColors] = useState({});
   const [cellInfoVisible, setCellInfoVisible] = useState(true);
+  const [cellIndexMapping, setCellIndexMapping] = useState(null);
+  const [getCellSetCellId, setGetCellSetCellId] = useState(null);
+  const [getEmbeddingCellIdx, setGetEmbeddingCellIdx] = useState(null);
```

Comment:
don't use "get" in the names, because it is confusing -- when I saw it, I thought that these are not variables, but rather getter methods. Also, what are these? What is `getCellSetCellId`, `cellIndexMapping` and `getEmbeddingCellIdx` and why do we need to have a state variable for each of these?

Reply:
`cellIndexMapping` is used to store the calculated mapping between cellId and the id that is used in the embedding. `getCellSetCellId` is a function that is used to map a cellId to the embedding id. `getEmbeddingCellIdx` is a function that is used to map back embedding id to the corresponding cellId.

I put all these values in a state so that they can all be calculated once in the `useEffect` body and then used for any interactions without calculating again.